src/main/java/DirectoryEncryptor/DirectoryProcessor.java

```diff
@@ -0,0 +1,26 @@
+package DirectoryEncryptor;
+
+import EncryptionAlgorithms.EncryptionAlgorithm;
+import EventsLogger.EncryptionLogEventsArgs;
+import EventsLogger.Events;
+import org.apache.logging.log4j.Level;
+
+import java.util.Observable;
+
+public abstract class DirectoryProcessor extends Observable implements IDirectoryProcessor {
+
```

Comment:
add here the common instances and functions of the sub classes (avoid duplicate code)